### PR TITLE
New Top-Level Navigation

### LINKED ISSUE #319: Top Level Navigation Options
### Epic

_No response_

### Description

With @iskerrett coming onboard, I wanted to do a review of our top-level navigation options. We have a lot of content that is only accessible via Footer links, and whilst some of this may not be too valuable, it does add flavour and personality to our site, e.g. handbook/about us/team.

#### Header

Screenshot 2023-01-10 at 14 02 55

#### Footer

Screenshot 2023-01-10 at 14 03 07

### Acceptance Criteria

- [ ] Agree upon most important navigation links
- [ ] Produce designs to show alternatives
- [ ] Develop/Deploy alternatives

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

https://user-images.githubusercontent.com/99246719/234575676-389ade8e-1a4f-498c-a8fa-3885474f2052.mov

- Complete overhaul of the top-level navigation
- Adds 11ty shortcode for re-use across nav items
- add purpose-built desktop/mobile "Resources" navigation. tried to build it as one instance and then re-style entirely via CSS but it just got too messy.
- Add "mobile ctas" section to include sign in and free trial CTAs to the mobile navigation menu
- Mobile navigation kicks in at tailwind's `md` width, and is full screen
- We now have animation of `max-height` on mobile & desktop. With the latter, it's a little jittery, I'm not totally happy with it, removing animation entirely causes some annoying mouse hover/delay issues, so better to keep it in.

Annoyances:
- The removal of the chevron means it's a little ambiguous which are clickable, and which are hover interactions. Have seen this on other sites too though (e.g. Stripe, mongo), so not a disaster
- Mobile navigation is driven by hover behaviour, rather than click interaction. On an actual tablet/mobile device, these are the same, hover = click. however, with an actual mouse, small window size is a little chaotic using hover. Ideally this would be click-driven (but is a lot more work, and annoying to do with Eleventy)

## Related Issue(s)

Closes #319 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes

### CODE REVIEW

**Comment by joepavitt** [comment]
Reviewers:
- @iskerrett for content/marketing eyes
- @Pezmc for code review

**Comment by joepavitt** [comment]
> Just one detail: on mobile, the chevron should rotate 180º or flip when the item is active/open.

done

File: .eleventy.js [hunk 1]
```diff
@@ -291,6 +291,32 @@ module.exports = function(eleventyConfig) {
         }
     });
 
+    eleventyConfig.addPairedShortcode("navoption", function(content, label, link, depth, icon, iconSolid, addClasses) {
+        var svg, iconSvg = '', classes, chevron
```
Comment (Pezmc) [inline .eleventy.js:295]:
```suggestion
        let svg, iconSvg = '', classes, chevron
```

File: src/_includes/components/icons/users.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6">
```
Comment (Pezmc) [inline src/_includes/components/icons/users.svg:1]: These SVG's could all benefit from lossless compression

Comment (Pezmc) [reply src/_includes/components/icons/users.svg]: Handled it via SVG go

**Review by Pezmc — APPROVED** [review]

**Comment by Pezmc** [comment]
@joepavitt Leaving the merge to you since this goes live instantly

**Pull request merged**